fix: LoadEventCompleted scene event not invoked when a client disconnects before finishing loading a scene [MTT-3396]

NoelStephensUnity opened the pull request:
This resolves #1902 where during a scene event (loading or unloading) if a client disconnects the event complete notification would wait until the scene loading timeout period before completing.
[MTT-3396](https://jira.unity3d.com/browse/MTT-3396)


## Changelog
- Fixed issue where one or more clients disconnecting during a scene event would cause `LoadEventCompleted` or `UnloadEventCompleted` to wait until the `NetworkConfig.LoadSceneTimeOut` period before being triggered.

## Testing and Documentation
- Includes integration tests.

NoelStephensUnity committed:
test

This allows me to test this bug to verify it is no longer happening.

NoelStephensUnity committed:
fix

This fixes the issue where if a client disconnects during a scene event it would not get completed.
This will also fix an exception on the client side where if the client was disconnected during a scene event while in the middle of loading or unloading a scene it will not try to continue processing the event.

NoelStephensUnity committed:
test

This includes the integration test that validates the fix.
This also includes some minor adjustments to the integration test scene handler to account for handling a client disconnecting in the middle of a scene loading event.

NoelStephensUnity committed:
Update CHANGELOG.md

NoelStephensUnity committed:
update

MTT-3396
Adding additional code that tracks the initial clients that were connected when the scene event starts in order to provide an accurate list of clients that timed out during scene loading.
Cleaned up some of the code in the integration test as well as added additional comments.

NoelStephensUnity committed:
Update CHANGELOG.md

LPLafontaineB commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/SceneEventProgress.cs:
```diff
@@ -61,9 +61,9 @@ internal class SceneEventProgress
 
         /// <summary>
-        /// The NetworkTime at the moment the scene switch was initiated by the server.
+        /// The local time when it the scene event was "roughly started"
```
```suggestion
        /// The local time when the scene event was "roughly started"
```

NoelStephensUnity committed:
Merge branch 'develop' into fix/mtt-3396-loadEventcompleted-not-invoked-when-client-disconnects

NoelStephensUnity committed:
Update com.unity.netcode.gameobjects/Runtime/SceneManagement/SceneEventProgress.cs

Co-authored-by: LPLafontaineB <louisphilippe.lb@unity.com>

NoelStephensUnity committed:
fix

Fixing an issue where if a virtual mac machine was running slower than expected it would time out on certain deferred message tests.  The primary area impacted was WaitForMessageOfTypeReceived which seemed to not use the DefaultTimeout value but 0.5s as the maximum time it would wait for the message.
This just changes the timeout to the Default timeout for WaitForMessageOfTypeReceived and I noticed WaitForMessageOfTypeHandled used 0.5s for the default time out period and set that to use the DefaultTimeout.

NoelStephensUnity committed:
fix

increasing the timeout period.

NoelStephensUnity committed:
Merge branch 'develop' into fix/mtt-3396-loadEventcompleted-not-invoked-when-client-disconnects

NoelStephensUnity committed:
Merge branch 'develop' into fix/mtt-3396-loadEventcompleted-not-invoked-when-client-disconnects
